File: Java-Core/src/main/java/presto/literal/BooleanLiteral.java
package presto.literal;

import presto.runtime.Context;
import presto.type.BooleanType;
import presto.type.IType;
import presto.value.Boolean;

public class BooleanLiteral extends Literal<Boolean> {

	public BooleanLiteral(String text) {
		super(text, Boolean.Parse(text));
	}
	
	@Override
	public IType check(Context context) {
		return BooleanType.instance();
	}
	
}

File: Java-CodeServer/src/main/resources/js/prompto/literal/BooleanLiteral.js
var Literal = require("./Literal").Literal;
var BooleanType = require("../type/BooleanType").BooleanType;
var Bool = require("../value/Bool").Bool;

function BooleanLiteral(text) {
	Literal.call(this, text, Bool.Parse(text));
	return this;
}

BooleanLiteral.prototype = Object.create(Literal.prototype);
BooleanLiteral.prototype.constructor = BooleanLiteral;

BooleanLiteral.prototype.check = function(context) {
	return BooleanType.instance;
};

exports.BooleanLiteral = BooleanLiteral;